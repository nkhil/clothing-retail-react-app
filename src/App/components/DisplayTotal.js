import React, { Component, Fragment } from "react"
import { formatPrice } from "../helpers/helperMethods"

class DisplayTotal extends Component {
  render = () => {
    return (
      <Fragment>
        <h3 id="total">Total: {formatPrice(this.props.calculateTotal())}</h3>
      </Fragment>
    )
  }
}

export default DisplayTotal
