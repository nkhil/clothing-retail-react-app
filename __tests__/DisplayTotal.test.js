import "@babel/polyfill"
import React from "react"
import DisplayTotal from "../src/App/components/DisplayTotal"
import renderer from "react-test-renderer"

require("react-router-dom")
jest.mock("react-router-dom")

const mockCalculateTotal = () => 2300

test("DisplayTotal renders correctly", () => {
  const displayTotal = renderer
    .create(<DisplayTotal calculateTotal={mockCalculateTotal} />)
    .toJSON()
  expect(displayTotal).toMatchSnapshot()
})
