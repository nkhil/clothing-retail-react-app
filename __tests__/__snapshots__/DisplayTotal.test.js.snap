// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayTotal renders correctly 1`] = `
<h3
  id="total"
>
  Total: 
  Â£23.00
</h3>
`;
